packages/toolpad-core/src/shared/navigation.tsx

```diff
@@ -165,3 +126,28 @@ export function getItemPath(navigation: Navigation, item: NavigationPageItem): s
   invariant(path, `Item not found in navigation: ${item.title}`);
   return path;
 }
+
+/**
+ * Checks if a specific navigation page item has the active page as a child item.
+ */
+export function hasSelectedNavigationChildren(
+  navigation: Navigation,
+  item: NavigationPageItem,
+  activePagePath: string,
+): boolean {
+  if (item.children) {
+    return item.children.some((nestedItem) => {
+      if (!isPageItem(nestedItem)) {
+        return false;
+      }
+
+      if (nestedItem.children) {
+        return hasSelectedNavigationChildren(navigation, nestedItem, activePagePath);
+      }
+
+      return activePagePath === getItemPath(navigation, nestedItem);
```

Comment:
To be sure, does this function work in cases where the nested navigation items are supplied via regex patterns? I might be missing something, but seems like the earlier function was checking for a `navigationItem.pattern` case and testing the path against the regex pattern in case it existed but I can't seem to find an analogous case here

Reply:
I believe the `activePagePath` that goes here as a parameter already accounts for pattern matching with the logic in the `useActivePage` hook. I'll check again just to be sure.

So the behavior should be that pattern matching in children items works, but only under the `segment`s in the parent items. If the parent also has a `pattern` it will probably be ignored in these cases and I think that's probably the way it should work.